### ISSUE TITLE
btrfs receive returned non-zero exit status 1

### ISSUE THREAD

**Issue by dev-zero** [masc3d/btrfs-sxbackup#3]
I get the following for every run after the first:

```
# btrfs-sxbackup --source-container-subvolume .sxbackup ssh://root@myhost.example.com:/home /var/backup/myhost.home
INFO btrfs-sxbackup v0.4.6
INFO Source :: Url [ssh://root@myhost.example.com:/home] snapshot container subvolume [/home/.sxbackup] keep [10]
INFO Destination :: Url [/var/backup/myhost.home] snapshot container subvolume [/var/backup/myhost.home/] keep [10]
INFO Preparing environment
INFO Source :: Retrieving snapshot names
INFO Destination :: Retrieving snapshot names
INFO Source :: Creating snapshot
INFO Transferring snapshot
ERROR Command '['bash', '-c', 'btrfs receive /var/backup/myhost.home']' returned non-zero exit status 1
ERROR Traceback (most recent call last):
  File "/usr/lib64/python3.3/site-packages/btrfs_sxbackup/__main__.py", line 100, in <module>
    backup.run()
  File "/usr/lib64/python3.3/site-packages/btrfs_sxbackup/Backup.py", line 436, in run
    raise subprocess.CalledProcessError(receive_returncode, receive_command, None)
subprocess.CalledProcessError: Command '['bash', '-c', 'btrfs receive /var/backup/myhost.home']' returned non-zero exit status 1
```

This is with the following versions:

* btrfs-sxbackup head 11996a1bcc
* btrfs-progs: 3.18, kernel: 3.18.2 (receiver)
* btrfs-progs: 3.17.3, kernel: 3.17.3 (sender)
* python: 3.4.1

the source is a simple subvolume (`/home`), the dest is a subvolume of a subvolume (`/var/backup/myhost.home`, `/var/backup`)

**Comment by masc3d** [comment]
I have seen send/receive issues with differing versions of btrfs-progs. can you check with same version on both systems?

**Comment by dev-zero** [comment]
same result with btrfs-progs 3.18.2 on both sides:

```
# btrfs --version
Btrfs v3.18.2
```

```
# btrfs --version
Btrfs v3.18.2
```

is there a way I can get the output of `btrfs receive`?

**Comment by masc3d** [comment]
works fine with 3.18.2 here, but I also do have kernel-3.18 on both sides. 
if there's any output I don't think it will help much, but you could remove "stdout=subprocess.PIPE" from Backup.py line 405 to see it on stdout.

**Comment by dev-zero** [comment]
Thanks. That's what I get:

```
It seems that you have changed your default subvolume or you specify other subvolume to
mount btrfs, try to remount this btrfs filesystem with fs tree, and run btrfs receive again!
```

**Comment by masc3d** [comment]
hm odd. it's not an issue with btrfs-sxbackup though. not sure if it helps, but I would try upgrading the sender to kernel-3.18, too. did you use btrfs-convert or mkfs.btrfs to create the fs?

**Comment by dev-zero** [comment]
... and it succeeds when I mount the default subvolume and point the destination there:

```
# btrfs-sxbackup --source-container-subvolume .sxbackup ssh://root@myhost.example.com:/home /mnt/sda/backup/myhost.home
INFO btrfs-sxbackup v0.4.6
INFO Source :: Url [ssh://root@myhost.example.com:/home] snapshot container subvolume [/home/.sxbackup] keep [10]
INFO Destination :: Url [/mnt/sda/backup/myhost.home] snapshot container subvolume [/mnt/sda/backup/myhost.home/] keep [10]
INFO Preparing environment
INFO Source :: Retrieving snapshot names
INFO Destination :: Retrieving snapshot names
INFO Source :: Creating snapshot
INFO Transferring snapshot
At snapshot temp
INFO Backup sx-20150131-200205-utc created successfully in 00:00:09
# ls /mnt/sda/backup/myhost.home/
sx-20150131-185236-utc  sx-20150131-200205-utc
```

**Comment by dev-zero** [comment]
`mkfs.btrfs` and I didn't change the default subvolume.
Please note: `/var/backup` is a subvolume in my case. Did you test it with such a scenario?

**Comment by masc3d** [comment]
yes, the destination should also be a subvolume. so did I understand correctly, it does not work if the receiver's subvolume is mounted with subvol=?

**Comment by dev-zero** [comment]
Exactly. When the receiver mountpoint is mounted using `subvol=` it does not work here.

**Comment by masc3d** [comment]
yes I can confirm this. getting the same error as soon as receiver's subvolume is not mounted via fs root. I will add this to the readme..

**Comment by masc3d** [comment]
btrfs receive (yet quite useful!) output is now included in exception/error message in >=0.5.0. thanks

**Comment by hayalci** [comment]
Can you say a few sentences in the readme about this? The error message is very cryptic ("try to remount this btrfs filesystem with fs tree" -- ??), it would help if we can include something like this:

>The destination subvolume must not be mounted by the subvol mount option, it should be a path
>within the default subvolume. Otherwise you get the following error: 
>"It seems that you have changed your default subvolume or you specify other subvolume to
>mount btrfs, try to remount this btrfs filesystem with fs tree, and run btrfs receive again!"


FYI: The commit introducing the error message is the following. I'll try to contact the author and ask for a more clear error message. https://www.mail-archive.com/linux-btrfs@vger.kernel.org/msg29800.html

**Comment by masc3d** [comment]
added to known limitations section in the readme, thanks.

**Comment by bmathieu33** [comment]
For the record it looks like it's fixed in btrfs-progs 4.1 (not tested myself):
https://btrfs.wiki.kernel.org/index.php/Changelog#btrfs-progs_4.1_.28Jun_2015.29
`receive: can work inside subvolumes`

**Comment by dan-damian** [comment]
I had the same issue but the problem was because the folder where I've configured the btrfs pull job wasn't mounted on a btrfs.
Let me describe more exactly:
1) init pull job in /mnt/btrfs-root folder
sudo /usr/local/bin/btrfs-sxbackup init ssh://root@remotehost:/myProject /mnt/btrfs-root/

2) first snapshot worked
sudo /usr/local/bin/btrfs-sxbackup run -m admin@somemail.com /mnt/btrfs-root/

3) second snapshot failed 
sudo /usr/local/bin/btrfs-sxbackup run -m admin@somemail.com /mnt/btrfs-root/ 
saying "It seems that you have changed your default subvolume or you specify other subvolume to
mount btrfs, try to remount this btrfs filesystem with fs tree, and run btrfs receive again!"

I've solved this by mounting /mnt/btrfs-root on btrfs.
Don't know how I've missed that.
